lib/updateDeps.js

```diff
@@ -0,0 +1,166 @@
+const { writeFileSync } = require("fs");
+const recognizeFormat = require("./recognizeFormat");
+const semver = require("semver");
+const getManifest = require("./getManifest");
+
+/**
+ * Resolve next package version.
+ *
+ * @param {Package} pkg Package object.
+ * @returns {string|undefined} Next pkg version.
+ * @internal
+ */
+const getNextVersion = (pkg) => {
+	const lastVersion = pkg._lastRelease && pkg._lastRelease.version;
+
+	return lastVersion && typeof pkg._nextType === "string" ? semver.inc(lastVersion, pkg._nextType) : "1.0.0";
+};
+
+/**
+ * Resolve package release type taking into account the cascading dependency update.
+ *
+ * @param {Package} pkg Package object.
+ * @param {string|undefined} bumpStrategy Dependency resolution strategy: override, satisfy, inherit.
+ * @param {string|undefined} releaseStrategy Release type triggered by deps updating: patch, minor, major, inherit.
+ * @param {Package[]} ignore=[] Packages to ignore (to prevent infinite loops).
+ * @returns {string|undefined} Resolved release type.
+ * @internal
+ */
+const resolveReleaseType = (pkg, bumpStrategy = "override", releaseStrategy = "patch", ignore = []) => {
+	// NOTE This fn also updates pkg deps, so it must be invoked anyway.
+	const dependentReleaseType = getDependentRelease(pkg, bumpStrategy, releaseStrategy, ignore);
+
+	// Release type found by commitAnalyzer.
+	if (pkg._nextType) {
+		return pkg._nextType;
+	}
+
+	if (!dependentReleaseType) {
+		return undefined;
+	}
+
+	pkg._nextType = releaseStrategy === "inherit" ? dependentReleaseType : releaseStrategy;
+
+	return pkg._nextType;
+};
+
+/**
+ * Get dependent release type by recursive scanning and updating its deps.
+ *
+ * @param {Package} pkg The package with local deps to check.
+ * @param {string} bumpStrategy Dependency resolution strategy: override, satisfy, inherit.
+ * @param {string} releaseStrategy Release type triggered by deps updating: patch, minor, major, inherit.
+ * @param {Package[]} ignore Packages to ignore (to prevent infinite loops).
+ * @returns {string|undefined} Returns the highest release type if found, undefined otherwise
+ * @internal
+ */
+const getDependentRelease = (pkg, bumpStrategy, releaseStrategy, ignore) => {
+	const severityOrder = ["patch", "minor", "major"];
+	const { localDeps, manifest = {} } = pkg;
+	const { dependencies = {}, devDependencies = {}, peerDependencies = {}, optionalDependencies = {} } = manifest;
+	const scopes = [dependencies, devDependencies, peerDependencies, optionalDependencies];
+	const bumpDependency = (scope, name, nextVersion) => {
+		const currentVersion = scope[name];
+		if (!nextVersion || !currentVersion) {
+			return;
```

Comment:
```suggestion
			return false;
```

As to not return `true | undefined` and instead just always return a boolean.

Reply:
Reasonable.